client/sites-dashboard/components/sites-table-row.tsx

```diff
@@ -142,7 +143,22 @@ export default function SitesTableRow( { site }: SiteTableRowProps ) {
 							href={ getDashboardUrl( site.slug ) }
 							title={ __( 'Visit Dashboard' ) }
 						>
-							<SiteIcon siteId={ site.ID } size={ 50 } />
+							<SiteThumbnail
+								mShotsUrl={ setmShotsUrl ? site.URL : undefined }
+								alt={ site.name }
+								bgColorImgUrl={ site.icon?.img }
+							>
+								{ site.icon ? (
+									<Image
+										src={ site.icon.img }
+										alt={ site.name }
+										style={ { height: '50px', width: '50px' } }
+										className={ undefined }
```

Comment:
It's a bit strange having to pass `undefined` here. I'm guessing it's because you got a TypeScript error?

`<Image>` is implemented in JS and TypeScript is doing its best to infer the prop types, but it's doing a bad job and gotten it wrong. You can fix this by adding this esoteric jsdoc to the `Image` function definition in `image/index.js`

```js
/**
 * @param {import('react').ImgHTMLAttributes} props
 */
export default function Image( { alt, className, src, ...restProps } ) {
```

This says that the `<Image>` component takes all the same prop types as the `<img>` element.

This fixes things up so that we can leave `className` blank (and it should improve the typing for everyone else using `<Image>` too!